feat: add delete offer endpoint with tests and constants

HijoDeHonor opened the pull request:
- Implemented DELETE endpoint '/(offers)/{IDOferta}' to delete an offer from the database once the post is completed in history
- Created function to execute after the exchange is finalized, deleting the corresponding offer from the database
- Added unit tests for the delete offer endpoint
- Added text constants for better maintainability
- Improved error handling
- Adjusted import paths for better organization

HijoDeHonor committed:
feat: add delete offer endpoint with tests and constants

- Implemented DELETE endpoint '/(offers)/{IDOferta}' to delete an offer from the database once the post is completed in history
- Created function to execute after the exchange is finalized, deleting the corresponding offer from the database
- Added unit tests for the delete offer endpoint
- Added text constants for better maintainability
- Improved error handling
- Adjusted import paths for better organization

PiewDev commented on src/offer/offerService.js:
```diff
@@ -7,4 +10,12 @@ export class OfferService {
   };
+
+  deleteOffer = async (id) => {
+    const isDelete = await this.offerRepository.deleteOffer(id);
```
isDelete implies that it is a boolean value and it is not

PiewDev replied:
If you return a boolean as suggested it would be fine

PiewDev commented on src/offer/offerService.js:
```diff
@@ -7,4 +10,12 @@ export class OfferService {
+    if (isDelete.success !== true) {
+      throw new FailedToDeleteError(FAILED_DELETING, OFFERS);
+    }
+    return isDelete;
```
It is not necessary to return anything, if it does not work it will throw an exception

PiewDev commented on src/offer/offerController.js:
```diff
@@ -1,14 +1,26 @@
+
+  deleteOffer = tryCatch(async (req, res) => {
+    const id = req.params.id;
+    if (!id) {
```
this check goes in the service

PiewDev commented on src/offer/offerRepository.js:
```diff
@@ -32,4 +32,28 @@ export class OfferRepository {
+        COMMIT;`
+        , [id, id, id]
+      );
+      return {
```
you could just return a bool

PiewDev reviewed (changes requested):
There are some structural changes to be made

HijoDeHonor committed:
refactor: update delete offer endpoint logic and adjust tests

- Refactored DELETE endpoint '/offers/{IDOferta}' to change the deletion logic
- Updated unit tests to reflect the new deletion logic

HijoDeHonor committed:
refactor: update delete offer endpoint logic and adjust tests

- Refactored DELETE endpoint '/(Tradinghall, Market?)/{IDOferta}' to change the deletion logic
- Updated unit tests to reflect the new deletion logic
- Improved error handling
- Added text constants for better maintainability
- Adjusted import paths for better organization
